test/runtests.jl

```diff
@@ -266,3 +266,18 @@ end
 
 # Check that printing to the default STDOUT doesn't fail
 JSON.print(["JSON.jl tests pass!"],1)
+
+# Tests for Macro strings
+
+let mstr_test = @compat Dict("a" => 1)
+    @test mstr_test == json"""{"a":1}"""
+    @test json"""{"x": 3}"""OrderedDict == DataStructures.OrderedDict{String,Any}([("x",3)])
```

Comment:
this syntax is strange... I'd prefer to just use `JSON.parse` if a special dict type is needed